aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -599,6 +599,98 @@ Tensor& diff_out(const Tensor& self, int64_t n, int64_t dim, const c10::optional
   }
 }
 
+void pre_check_gradient(const Tensor& self, IntArrayRef dim, int64_t spacing_size, int64_t edge_order, bool spacing_existence, bool dim_existence) {
+  // Helper for gradient function to make sure input data satisfies prerequisites
+  TORCH_CHECK(self.scalar_type() != ScalarType::Byte, "torch.gradient does not support uint8 input.");
+  if (spacing_existence && !dim_existence) {
+    TORCH_CHECK( spacing_size == 1 || spacing_size == self.dim(), "torch.gradent expected spacing has size of 1 or ", self.dim(), " or not given at all when dim is not provided." );
+  }
+  TORCH_CHECK( spacing_size == dim.size(), "torch.gradent expected its spacing and dim arguments to have the same length, but got a spacing argument of length ", spacing_size, " and a dim argument of length ", dim.size(), "." );
+  TORCH_CHECK( edge_order == 1, "torch.gradient only supports edge_order=1 currently. See discussion : https://github.com/pytorch/pytorch/issues/56036");
+  dim_list_to_bitset(dim, self.dim());
```

Comment:
to request support for more edge_orders please file an issue here : https://github.com/pytorch/pytorch/issues/new?assignees=&labels=&template=feature-request.md

Reply:
in the comment to add see discussion :)